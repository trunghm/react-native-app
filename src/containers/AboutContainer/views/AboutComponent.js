import React, {Component} from 'react'
import {View, ImageBackground, Image, Platform, Text} from 'react-native';

import styles from './styles';
import I18n from '../../../I18n/I18n';

export default class AboutComponent extends Component {
  constructor (props) {
    super(props)
    this.state = {

    }
  }




  render () {
    return (
      <View style={styles().container}>
        <Text style={styles().text}>{I18n.t('about_page.title_page')} </Text>
      </View>
    )
  }
}

