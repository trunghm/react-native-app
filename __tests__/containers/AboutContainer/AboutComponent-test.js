/**
 * @format
 */

import 'react-native';
import React from 'react';
import AboutComponent from '../../../src/containers/AboutContainer/views';

// Note: test renderer must be required after react-native.
import {mount, shallow} from 'enzyme';
import renderer from  "react-test-renderer";
import EStyleSheet from 'react-native-extended-stylesheet';


const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
  EStyleSheet.build({
  });
});

afterAll(() => {
  console.error = originalError;
});

describe('Testing  AboutComponent component', () => {

  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <AboutComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('renders as expected', () => {
    const wrapper = mount(
      <AboutComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

});
