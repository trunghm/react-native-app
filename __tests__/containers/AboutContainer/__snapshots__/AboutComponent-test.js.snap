// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  AboutComponent component renders as expected 1`] = `ReactWrapper {}`;

exports[`Testing  AboutComponent component renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
      }
    }
  >
    About
     
  </Text>
</View>
`;
